web/frontend/src/components/slam-2d-render.vue

```diff
@@ -63,25 +72,96 @@ controls.enableRotate = false;
 
 const raycaster = new MouseRaycaster({ camera, renderer, recursive: false });
 
-raycaster.on('click', (event) => {
+raycaster.on('click', (event: THREE.Event) => {
   const [intersection] = event.intersections as THREE.Intersection[];
   if (intersection && intersection.point) {
-    console.log(intersection.point);
+    emit('click', intersection.point);
   }
 });
 
-const markerSize = 0.5;
-const marker = new THREE.Mesh(
-  new THREE.PlaneGeometry(markerSize, markerSize).rotateX(-Math.PI / 2),
-  new THREE.MeshBasicMaterial({ color: 'red' })
-);
-marker.name = 'Marker';
-// This ensures the robot marker renders on top of the pointcloud data
-marker.renderOrder = 999;
+const guiData = {
+    drawFillShapes: true,
+    drawStrokes: true,
+    fillShapesWireframe: false,
+    strokesWireframe: false,
+  };
+/*
+ * svgLoader example for webgl:
+ * https://github.com/mrdoob/three.js/blob/master/examples/webgl_loader_svg.html
+ */
+//
+const svgLoader = new SVGLoader(); 
+const makeMarker = async (url : string, name: string, scalar: number) => {
+  const data = await svgLoader.loadAsync(url);
+
+  const { paths } = data!;
+
+  const group = new THREE.Group();
+  group.scale.multiplyScalar(scalar);
+  group.position.set(-70, 0, 70) // why do we do this? - do we need it?
+  group.scale.y *= -1; // why do we do this? - do we need it?
+
+  for (const path of paths) {
+
+    const fillColor = path!.userData!.style.fill;
+
+    if (guiData.drawFillShapes && fillColor !== undefined && fillColor !== 'none') {
+      const material = new THREE.MeshBasicMaterial({
+        color: new THREE.Color().setStyle(fillColor)
+          .convertSRGBToLinear(),
+        opacity: path!.userData!.style.fillOpacity,
+        transparent: true,
+        side: THREE.DoubleSide,
```

Comment:
Why is this DoubleSide? since this view is only rendered top-down I'd assume that this should be `THREE.FrontSide`

Reply:
setting to `THREE.FrontSide` does not render the base marker correctly.
<img width="226" alt="Screen Shot 2023-04-18 at 10 15 03 AM" src="https://user-images.githubusercontent.com/46872531/232854055-7056bead-4e5c-4bf5-8b8b-fd3eb4d8b04f.png">

Reply:
oh, that may be because of
```
  group.scale.y *= -1; // why do we do this? - do we need it?

Reply:
same outcome

Reply:
:| nvm, will address later, it's not a big deal